File: rpg.js
// Define a player object with name, hit points, and attack strength
const player = {
    name: "Hero",
    hitPoints: 100,
    attackStrength: 10
  };
  
  // Define a monster object with name, hit points, and attack strength
  const monster = {
    name: "Dragon",
    hitPoints: 150,
    attackStrength: 15
  };
  
  // Define a function to attack the monster
  const attackMonster = () => {
    // Reduce the monster's hit points by the player's attack strength
    monster.hitPoints -= player.attackStrength;
  
    // Check if the monster is dead
    if (monster.hitPoints <= 0) {
      console.log(`The ${monster.name} has been defeated!`);
    } else {
      console.log(`The ${monster.name} has ${monster.hitPoints} hit points remaining.`);
    }
  };
  
  // Attack the monster three times
  attackMonster();
  attackMonster();
  attackMonster();
  

File: rpg.rs
struct Player {
    name: String,
    hit_points: u32,
  }
  
  struct Monster {
    name: String,
    hit_points: u32,
  }
  
  fn attack_monster(player: &mut Player, monster: &mut Monster) {
    monster.hit_points -= player.hit_points;
  
    if monster.hit_points <= 0 {
      println!("{} has defeated the {}!", player.name, monster.name);
    } else {
      println!("{} has {} hit points remaining.", player.name, player.hit_points);
      println!("{} has {} hit points remaining.", monster.name, monster.hit_points);
    }
  }
  
  fn main() {
    let mut player = Player {
      name: "Hero".to_string(),
      hit_points: 100,
    };
  
    let mut monster = Monster {
      name: "Dragon".to_string(),
      hit_points: 150,
    };
  
    attack_monster(&mut player, &mut monster);
    attack_monster(&mut player, &mut monster);
    attack_monster(&mut player, &mut monster);
  }
  